### ISSUE TITLE
InternVL3-2B runtime issue on AX650 (ax-llm)

### ISSUE THREAD

**Issue by SrivathsaTirumala** [AXERA-TECH/ax-llm#33]
## Environment

* AX650N
* Pulsar2 5.2
* InternVL3-2B
* ax-llm runtime

## What works

* `llm_build` conversion succeeds
* Vision encoder + projector exported successfully
* Vision AXModel runs correctly
* Decoder AXModels load successfully
* Prefill + decode loops execute without crashes

## Problem

Generated output is corrupted:

```text id="rgq7o6"
answer >> ������������
```

## Findings

InternVL3 internally uses Qwen2 architecture.

Decoder output ordering observed:

```text id="uqx8is"
outputs[0] -> K cache
outputs[1] -> V cache
outputs[2] -> hidden states
```

Hidden states shape:

```text id="snb67y"
(1, N, 1536)
```

KV cache shape:

```text id="hiv3mk"
(1, N, 256)
```

Also, during first prefill slice:

* model expects `(1,1,1536)` dummy K/V tensors
* not KV-dim `(1,1,256)`

Runtime now executes fully, but decoded tokens are invalid.

Suspected issue:

* postprocess AXModel output ordering/logits selection differs from Gemma/Llama handling in current `infer_func.py`.

Attached files:

* `internvl3_infer.py`
* `infer_func.py`

Could you clarify:

1. Expected postprocess output ordering for Qwen2/InternVL models
2. Whether InternVL3 is officially supported in ax-llm runtime

Thanks.

[internvl3_infer.py](https://github.com/user-attachments/files/27696957/internvl3_infer.py)

[infer_func.py](https://github.com/user-attachments/files/27696960/infer_func.py)

**Comment by skylake5200** [comment]
Hi @SrivathsaTirumala 

Unfortunately, InternVL3-2B​ still relies on the legacy inference method and does not currently support direct inference via axllm. If you require support for this specific method, please contact our support team to facilitate further assistance. Alternatively, we recommend trying our newly uploaded models, which are fully compatible with axllm.
